File: src/main/java/com/gmail/borlandlp/minigamesdtools/activepoints/StaticPointsCache.java
package com.gmail.borlandlp.minigamesdtools.activepoints;

import org.bukkit.Location;

import java.util.HashMap;
import java.util.Map;

public class StaticPointsCache {
    private Map<String, ActivePoint> pointBlocksCache = new HashMap<>();//(String) worldName|coordX|coordY|coordZ -> ActivePoint

    public void remove(Location location) {
        pointBlocksCache.remove(location2StringF(location));
    }

    public void add(Location location, ActivePoint activePoint) {
        pointBlocksCache.put(location2StringF(location), activePoint);
    }

    public ActivePoint get(Location location) {
        if(location == null) {
            return null;
        }

        return pointBlocksCache.get(location2StringF(location));
    }

    public boolean contains(Location location) {
        return pointBlocksCache.containsKey(location2StringF(location));
    }

    private String location2StringF(Location location) {
        return location.getWorld().getName() + "|" +
                location.getBlockX() + "|" +
                location.getBlockY() + "|" +
                location.getBlockZ();
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/activepoints/StaticPointsCache.kt
package com.gmail.borlandlp.minigamesdtools.activepoints

import org.bukkit.Location
import java.util.*

class StaticPointsCache {
    /* (String) worldName|coordX|coordY|coordZ -> ActivePoint */
    private val pointBlocksCache: MutableMap<String, ActivePoint> = mutableMapOf()

    fun remove(location: Location) {
        pointBlocksCache.remove(location2StringF(location))
    }

    fun add(location: Location, activePoint: ActivePoint) {
        pointBlocksCache[location2StringF(location)] = activePoint
    }

    operator fun get(location: Location): ActivePoint {
        return pointBlocksCache[location2StringF(location)]!!
    }

    operator fun contains(location: Location): Boolean {
        return pointBlocksCache.containsKey(location2StringF(location))
    }

    private fun location2StringF(location: Location): String {
        return "${location.world.name}|${location.blockX}|${location.blockY}|${location.blockZ}"
    }
}